src/App.jsx

```diff
@@ -14,45 +15,108 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
-
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+function getPreparedGoods(goodsFrom, sortField, reverse) {
+  const goods = [...goodsFrom];
+
+  if (sortField) {
+    goods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_NAME:
+          return a.localeCompare(b);
+
+        case SORT_FIELD_LENGTH:
+          return a.length - b.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    goods.reverse();
+  }
+
+  return goods;
+}
+
+function getReset(sortField, reverse) {
+  return sortField !== '' || reverse;
+}
+
+const SORT_FIELD_NAME = 'name';
+const SORT_FIELD_LENGTH = 'length';
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [reverse, setReversed] = useState(false);
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, reverse);
+  const reset = getReset(sortField, reverse);
+
+  const resetSort = () => {
+    setReversed(false);
+    setSortField('');
+  };
+
+  return (
+    <>
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={
+              sortField === SORT_FIELD_NAME
+                ? 'button is-info'
+                : 'button is-info is-light'
+            }
+            onClick={() => setSortField(SORT_FIELD_NAME)}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={
+              sortField === SORT_FIELD_LENGTH
+                ? 'button is-info'
+                : 'button is-info is-light'
+            }
+            onClick={() => setSortField(SORT_FIELD_LENGTH)}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={reverse ? 'button is-info' : 'button is-info is-light'}
+            onClick={() => setReversed(!reverse)}
+          >
```

Comment:
Create separate handlers instead of using anonymous funcstions.
For sorting it will looks like this one
```
const setSort = (field) => () => {
  setSortField(field)
}

const onReverse = () => {
  setReversed(!isReversed);
};
```
```suggestion
            onClick={setSort(SORT_FIELD_NAME)}
          >
            Sort alphabetically
          </button>

          <button
            type="button"
            className={
              sortField === SORT_FIELD_LENGTH
                ? 'button is-info'
                : 'button is-info is-light'
            }
            onClick={setSort(SORT_FIELD_LENGTH)}
          >
            Sort by length
          </button>

          <button
            type="button"
            className={reverse ? 'button is-info' : 'button is-info is-light'}
            onClick={onReverse}
          >
```